### PR TITLE
Nexus work directory

### PR DESCRIPTION
**Pull request by jdillon**
Replaces `../sonatype-work/nexus` with `$install/data`.

This is needed for simplification installer, as well as other purposes to default some content into the nexus-work directory.

This _mostly_ normalizes the writable directories that nexus requires to function to just `$install/data` and only 1 place to change this if user wishes to use another directory.

related:
https://github.com/sonatype/nexus-pro/pull/548
https://github.com/sonatype/nexus-bundles/pull/160

outdated but also related:
https://github.com/sonatype/nexus-bundles/pull/109

### CODE REVIEW

**Comment by jdillon** [comment]
Using https://github.com/sonatype/nexus-oss/pull/1350 for now, will drop this after that is merged and bring back PR which addresses sonatype-work for post M4 change

File: assemblies/nexus-base-template/src/main/resources/overlay/etc/nexus.properties [hunk 1]
```diff
@@ -4,7 +4,7 @@ application-host=0.0.0.0
 nexus-context-path=/
 
 # Nexus section
-nexus-work=${nexus-base}/../sonatype-work/nexus
+nexus-work=${karaf.data}
```
Comment (mcculls) [inline assemblies/nexus-base-template/src/main/resources/overlay/etc/nexus.properties]: `${karaf.data}/nexus` might be clearer and avoid potential clashes with karaf directories going forwards

**Comment by mcculls** [comment]
+1 but I'd prefer to see `${karaf.data}/nexus` as the working directory

**Comment by jdillon** [comment]
@mcculls I did consider that, but didn't see too much overlap to warrant the need for another sub-directory

**Comment by jdillon** [comment]
@mcculls here is what it looks like with data/nexus:

```
[ jason Lust nexus-oss ] ls -l target/nexus-base-template-3.0.0-SNAPSHOT/data/
total 8
drwxr-xr-x 240 jason staff 8160 Jul 13 13:35 cache
drwxr-xr-x   2 jason staff   68 Jul 13 13:35 generated-bundles
-rw-r--r--   1 jason staff   36 Jul 13 13:35 lock
drwxr-xr-x   3 jason staff  102 Jul 13 13:35 log
drwxr-xr-x  10 jason staff  340 Jul 13 13:35 nexus
-rw-r--r--   1 jason staff    5 Jul 13 13:35 port
drwxr-xr-x   5 jason staff  170 Jul 13 13:35 tmp
[ jason Lust nexus-oss ] ls -l target/nexus-base-template-3.0.0-SNAPSHOT/data/nexus/
total 4
drwxr-xr-x 3 jason staff 102 Jul 13 13:35 blobs
drwxr-xr-x 7 jason staff 238 Jul 13 13:35 db
drwxr-xr-x 3 jason staff 102 Jul 13 13:35 elasticsearch
drwxr-xr-x 5 jason staff 170 Jul 13 13:35 etc
drwxr-xr-x 3 jason staff 102 Jul 13 13:35 keystores
drwxr-xr-x 4 jason staff 136 Jul 13 13:35 log
-rw-r--r-- 1 jason staff  36 Jul 13 13:35 nexus.lock
drwxr-xr-x 3 jason staff 102 Jul 13 13:35 orient
```

Note there are 2 different log dirs.  And locks in 2 places, I think this may be even more confusing that concerns about overlap of the few dirs that karaf uses.

**Comment by mcculls** [comment]
OK, based on the above I'm fine with using `${karaf.data}`

**Comment by adreghiciu** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**